.github/workflows/mlir-tensorrt-ci.yml

```diff
@@ -0,0 +1,227 @@
+name: MLIR-TensorRT CI
+
+on:
+  pull_request:
+    branches:
+      - main
+    #types: [synchronize, opened, reopened, ready_for_review]
+    #paths: ['mlir-tensorrt/**']
+
+env:
+  DEFAULT_IMAGE: ghcr.io/nvidia/tensorrt-incubator/mlir-tensorrt:cuda12.5-ubuntu-llvm17
+  ADHOC_TEST_IMAGE: adhoc-test-image:cuda12.5-ubuntu-llvm17
```

Comment:
What is 'ADHOC_TEST_IMAGE' for?

Reply:
This is tag for the docker image that will be created on fly, if any files that affect docker image are changed. In our case, they are the following.
```
- 'mlir-tensorrt/build_tools/docker/Dockerfile'
- 'mlir-tensorrt/python/requirements.txt'
- 'mlir-tensorrt/python/requirements-dev.txt'
```
If above files are changed and we are still testing with `DEFAULT_IMAGE`, there is chance that tests will fail (for example, if new python dependency is added). CI to generate a new `DEFAULT_IMAGE` if changes are in the above file runs after PR is merged in main.

If none of the above files is changed, we use `DEFAULT_IMAGE`.

Reply:
I would avoid rebuilding the entire container just for `requirements.txt` changes. Those you can `pip install` on the fly before running the build.

Reply:
Done.